### PR TITLE
[Validation] Go test automation Framework v2.6

### PR DESCRIPTION
**Pull request by igomez06**
Added test cases for provisioning custom RKE2 clusters. Also test cases for 1 node all roles and 3 nodes 1 role per node. Implemented aws package build on top of s3/ec2 package for provisioning ec2 instances. Also added DigitalOceanCredentialConfig to pkg/apis/management.cattle.io/v3machine_types.go and regenerated the types.

Have Dynamic test cases where a end user can define nodes and the node roles.

### CODE REVIEW

**Comment by igomez06** [comment]
https://github.com/rancher/rancher/pull/33901

**Comment by igomez06** [comment]
https://github.com/rancher/qa-tasks/issues/110

**Review by git-ival — DISMISSED** [review]
lgtm

File: tests/go-validation/cluster/cluster.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+}
+
+func (c *Cluster) DeleteCluster(clusterName string) error {
+	ctx := context.Background()
+	err := c.Delete(ctx, clusterName, metav1.DeleteOptions{})
+	return err
+}
+
+func ClusterCleanup(ec2Client *aws.EC2Client, clusters []*Cluster, clusterNames []string, nodes []*aws.EC2Node) error {
+	if ec2Client != nil {
```
Comment (git-ival) [inline tests/go-validation/cluster/cluster.go:145]: Shouldn't we remove the Clusters before the Nodes?

Comment (igomez06) [reply tests/go-validation/cluster/cluster.go:145]: Yeah that makes more sense. I'll make the change

Comment (igomez06) [reply tests/go-validation/cluster/cluster.go]: Fixed

**Pull request closed without merging**